ui/src/album/GridMenu.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react'
+import IconButton from '@material-ui/core/IconButton'
+import Menu from '@material-ui/core/Menu'
+import MenuItem from '@material-ui/core/MenuItem'
+import MoreVertIcon from '@material-ui/icons/MoreVert'
+import { useDispatch } from 'react-redux'
+import { playAlbum, shuffleAlbum } from '../audioplayer'
+import { useGetList } from 'react-admin'
+
+const GridMenu = (props) => {
+  const [anchorEl, setAnchorEl] = React.useState(null)
+  const dispatch = useDispatch()
+  const { ids, data, loading, error } = useGetList(
+    'albumSong',
+    {},
+    { field: 'trackNumber', order: 'ASC' },
+    { album_id: props.id }
+  )
+
+  if (loading) {
+    return (
+      <IconButton>
+        <MoreVertIcon />
+      </IconButton>
+    )
+  }
+
+  if (error) {
+    return (
+      <IconButton>
+        <MoreVertIcon />
+      </IconButton>
+    )
+  }
+
+  const options = [
+    [1, 'Play'],
```

Comment:
Maybe a different structure here could improve readability. What about:
```js
const options = {
  play: { 
    label: translate('resources.album.actions.playAll'), // From the translations file. Check the `en.js` and the `AlbumActions.js` files
    action: (data, id) => (playAlbum(id, data))
  },
  shuffle: ....
}
```

This way the `MenuItem`s rendering would be something like:
```jsx
{Onject.keys(options).map((key) => (
   <MenuItem value={key} key={key} onClick={handleItemClick}>
      {option[key].label}
   </MenuItem>
))}
```

and the `handleItemClick` would have something like:
```js
const key = e.target.getAttribute('value')
dispatch(options[key].action(data, ids[0])
```
instead of the cascade of ifs. We could then add new actions to the options object without changing other parts of the code. 

What do you think? (Keep in mind that the code above is not complete and may have typos and syntax errors)

Reply:
Yes it could be more efficient.I will go through the comments and make modifications as suggeted.